internal/common/customplanmodifier/non_updatable.go

```diff
@@ -0,0 +1,36 @@
+package customplanmodifier
+
+import (
+	"context"
+	"fmt"
+
+	planmodifier "github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+)
+
+func NonUpdatableStringAttributePlanModifier() planmodifier.String {
+	return &nonUpdatableStringAttributePlanModifier{}
+}
+
+type nonUpdatableStringAttributePlanModifier struct {
+}
+
+func (d *nonUpdatableStringAttributePlanModifier) Description(ctx context.Context) string {
+	return "Ensures that update operations fails when updating an attribute."
+}
+
+func (d *nonUpdatableStringAttributePlanModifier) MarkdownDescription(ctx context.Context) string {
+	return d.Description(ctx)
```

Comment:
knit: we're giving preference to MarkdownDescription, i know it's the same behavior but to be consitent I would define the string in MarkdownDescription